pkg/kubectl/cmd/resource/get.go

```diff
@@ -320,145 +345,51 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
+	options.HumanPrintFlags.ShowLabels = &showKind
+
+	printer, matches, err := options.HumanPrintFlags.ToPrinter(options.OutputFormat)
+	if !matches {
+		return fmt.Errorf("unable to match a printer to handle current print options")
+	}
 	if err != nil {
 		return err
 	}
-	var sorter *printers.RuntimeSort
-	if options.Sort && len(objs) > 1 {
-		// TODO: questionable
-		if sorter, err = printers.SortObjects(cmdutil.InternalVersionDecoder(), objs, sorting); err != nil {
-			return err
-		}
-	}
-
-	// use the default printer for each object
-	printer = nil
-	var lastMapping *meta.RESTMapping
-	w := printers.GetNewTabWriter(options.Out)
-
-	useOpenAPIPrintColumns := cmdutil.GetFlagBool(cmd, useOpenAPIPrintColumnFlagLabel)
-
-	showKind := options.ShowKind || resource.MultipleTypesRequested(args) || cmdutil.MustPrintWithKinds(objs, infos, sorter)
-
-	noHeaders := cmdutil.GetFlagBool(cmd, "no-headers")
-	for ix := range objs {
-		var mapping *meta.RESTMapping
-		var original runtime.Object
-		var info *resource.Info
-		if sorter != nil {
-			info = infos[sorter.OriginalPosition(ix)]
-			mapping = info.Mapping
-			original = info.Object
-		} else {
-			info = infos[ix]
-			mapping = info.Mapping
-			original = info.Object
-		}
-
-		// if dealing with a table that has no rows, skip remaining steps
-		// and avoid printing an unnecessary newline
-		if table, isTable := info.Object.(*metav1beta1.Table); isTable {
-			if len(table.Rows) == 0 {
-				continue
-			}
-		}
-
-		if shouldGetNewPrinterForMapping(printer, lastMapping, mapping) {
-			if printer != nil {
-				w.Flush()
-			}
-
-			printWithNamespace := options.AllNamespaces
-			if mapping != nil && mapping.Scope.Name() == meta.RESTScopeNameRoot {
-				printWithNamespace = false
-			}
-
-			printOpts := cmdutil.ExtractCmdPrintOptions(cmd, printWithNamespace)
-			// if cmd does not specify output format and useOpenAPIPrintColumnFlagLabel flag is true,
-			// then get the default output options for this mapping from OpenAPI schema.
-			if !cmdSpecifiesOutputFmt(cmd) && useOpenAPIPrintColumns {
-				updatePrintOptionsForOpenAPI(f, mapping, printOpts)
-			}
-
-			printer, err = cmdutil.PrinterForOptions(printOpts)
-			if err != nil {
-				if !errs.Has(err.Error()) {
-					errs.Insert(err.Error())
-					allErrs = append(allErrs, err)
-				}
-				continue
-			}
-
-			// TODO: this doesn't belong here
-			// add linebreak between resource groups (if there is more than one)
-			// skip linebreak above first resource group
-			if lastMapping != nil && !noHeaders {
-				fmt.Fprintf(options.ErrOut, "%s\n", "")
-			}
 
-			lastMapping = mapping
-		}
+	// TODO(juanvallejo): handling this inside of humanprinter_flags.go causes an import cycle
+	// figure out how to attach these handlers without resorting to each command having to do it.
+	printersinternal.AddHandlers(printer.(*printers.HumanReadablePrinter))
```

Comment:
So far everything works as expected, however I'm stuck having to add print handlers to the printer here, since adding them in `humanreadable_flags.go` causes an import cycle.

Solution would be to move the handlers out of `pkg/printers/internalversion` and into `pkg/printers`

Reply:
> Solution would be to move the handlers out of pkg/printers/internalversion and into pkg/printers

Wrong solution.  Make a new package.